app/vmagent/remotewrite/remotewrite.go

```diff
@@ -168,29 +168,47 @@ func Init() {
 			case <-stopCh:
 				return
 			}
-			configReloads.Inc()
+			relabelConfigReloads.Inc()
 			logger.Infof("SIGHUP received; reloading relabel configs pointed by -remoteWrite.relabelConfig and -remoteWrite.urlRelabelConfig")
 			rcs, err := loadRelabelConfigs()
 			if err != nil {
-				configReloadErrors.Inc()
-				configSuccess.Set(0)
+				relabelConfigReloadErrors.Inc()
+				relabelConfigSuccess.Set(0)
 				logger.Errorf("cannot reload relabel configs; preserving the previous configs; error: %s", err)
 				continue
 			}
-
 			allRelabelConfigs.Store(rcs)
-			configSuccess.Set(1)
-			configTimestamp.Set(fasttime.UnixTimestamp())
+			relabelConfigSuccess.Set(1)
+			relabelConfigTimestamp.Set(fasttime.UnixTimestamp())
 			logger.Infof("Successfully reloaded relabel configs")
+
+			hasErrors := false
+			for _, rwctxs := range rwctxsMap {
+				for _, rwctx := range rwctxs {
+					err = rwctx.reloadStreamAggr()
+					if err != nil {
+						hasErrors = true
+					}
+				}
+			}
+			for _, rwctx := range rwctxsDefault {
+				err = rwctx.reloadStreamAggr()
+				if err != nil {
+					hasErrors = true
+				}
+			}
+			if !hasErrors {
```

Comment:
I'm not sure if we should allow re-loading configs for one URL when config reload for another URL failed.
While it looks cool, but it is not consistent with relableingRules reload and moves responsibility for logging and metrics from the caller to the function it calls. This is also inconsistent with relabeling.
For example, `reloadStreamAggr` returns an `error`. But no-one ever reads that error. That's a sign of inconsistency.

Reply:
fixed